src/bootstrap/dist.rs

```diff
@@ -87,6 +87,39 @@ impl Step for Docs {
     }
 }
 
+#[derive(Debug, PartialOrd, Ord, Copy, Clone, Hash, PartialEq, Eq)]
+pub struct JsonDocs {
+    pub host: TargetSelection,
+}
+
+impl Step for JsonDocs {
+    type Output = Option<GeneratedTarball>;
+    const DEFAULT: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        let alias = "rust-json-docs";
+        let default = should_build_extended_tool(&run.builder, alias);
+        run.alias(alias).default_condition(default)
```

Comment:
I think it's ok to build this whenever we build docs; I don't know why you'd want one without the other. They should still be separate dist components though, so you can test your changes without having to wait for `rust-docs` to build (it takes a long time to run).
```suggestion
        run.alias("rust-json-docs").default_condition(run.builder.config.docs)
```

Reply:
I chose to add a separate configuration option to make it easy to switch the JSON part off in case there are ICEs when building the JSON docs for the compiler crates. But it built just fine when I tried it locally, so it was probably overly paranoid on my side 😁